core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -0,0 +1,137 @@
+        );
+    }
+
+    /**
+     * Create a sidekick container and attach it to another container. The VNC output of that container will be recorded.
+     */
+    public VncRecordingContainer(@NonNull Network network, @NonNull String targetNetworkAlias) throws IllegalStateException {
+        super(TestcontainersConfiguration.getInstance().getVncRecordedContainerImage());
+
+        this.targetNetworkAlias = targetNetworkAlias;
+        withNetwork(network);
+
+        waitingFor(new AbstractWaitStrategy() {
+
+            @Override
+            protected void waitUntilReady() {
+                try {
+                    Unreliables.retryUntilTrue((int) startupTimeout.toMillis(), TimeUnit.MILLISECONDS, () -> {
+                        CountDownLatch latch = new CountDownLatch(1);
+
+                        FrameConsumerResultCallback callback = new FrameConsumerResultCallback() {
+                            @Override
+                            public void onNext(Frame frame) {
+                                if (frame != null && new String(frame.getPayload()).contains("Connected")) {
+                                    latch.countDown();
+                                }
+                            }
+                        };
+
+                        try (
+                                Closeable __ = dockerClient.logContainerCmd(containerId)
+                                        .withFollowStream(true)
+                                        .withSince(0)
+                                        .withStdErr(true)
+                                        .exec(callback)
+                        ) {
+                            return latch.await(1, TimeUnit.SECONDS);
+                        }
+                    });
+                } catch (TimeoutException e) {
+                    throw new ContainerLaunchException("Timed out waiting for log output", e);
+                }
+            }
+        });
+    }
+
+    public VncRecordingContainer withVncPassword(@NonNull String vncPassword) {
+        this.vncPassword = vncPassword;
+        return this;
+    }
+
+    public VncRecordingContainer withVncPort(int vncPort) {
+        this.vncPort = vncPort;
+        return this;
+    }
+
+    public VncRecordingContainer withFrameRate(int frameRate) {
+        this.frameRate = frameRate;
+        return this;
+    }
+
+    @Override
+    protected void configure() {
+        withCreateContainerCmdModifier(it -> it.withEntrypoint("/bin/sh"));
+        setCommand(
+                "-c",
+                "echo '" + Base64.encodeBase64String(vncPassword.getBytes()) + "' | base64 -d > /vnc_password && " +
+                "flvrec.py -o " + RECORDING_FILE_NAME + " -d -r " + frameRate + " -P /vnc_password " + targetNetworkAlias + " " + vncPort
+        );
+    }
+
+    @SneakyThrows
+    public InputStream streamRecord() {
```

Comment:
the whole point is having "volume-less" recorder because:
1) for some reason, on Mac, files were incomplete (read - empty) once a recording is flushed to the FS ( See #466 )
2) allows Docker to be on another host. Not-so-critical for the integration testing, but I've received some feedback about CI setups for functional/e2e testing, people really run their Docker hosts "somewhere else"

Reply:
2 is definitely an awesome reason to do this 👍